src/substrate/processor.ts

```diff
@@ -53,7 +53,18 @@ export class Processor extends IEventProcessor<ApiPromise, Block> {
     };
 
     const events = await Promise.all(block.events.map(({ event }) => applyFilters(event)));
-    const extrinsics = await Promise.all(block.extrinsics.map((extrinsic) => applyFilters(extrinsic)));
-    return [...events, ...extrinsics].filter((e) => !!e); // remove null / unwanted events
+
+    // remove unsuccessful extrinsics, only keep extrinsics that map to ExtrinsicSuccess events
+    // cf: https://polkadot.js.org/docs/api/cookbook/blocks#how-do-i-map-extrinsics-to-their-events
+    const successfulExtrinsics = block.extrinsics.filter((_extrinsic, index) => {
```

Comment:
I don't know how stable or fragile this is...

Reply:
Quite stable from my testing